src/runtime_src/tools/xclbinutil/XclBinClass.cxx

```diff
@@ -587,6 +587,23 @@ XclBin::addSection(Section* _pSection) {
   m_xclBinHeader.m_header.m_numSections = (uint32_t) m_sections.size();
 }
 
+void 
+XclBin::addReplaceSection(ParameterSectionData &_PSD)
+{
+  enum axlf_section_kind eKind;
+  if (Section::translateSectionKindStrToKind(_PSD.getSectionName(), eKind) == false) {
+    std::string errMsg = XUtil::format("ERROR: Section '%s' isn't a valid section name.", _PSD.getSectionName().c_str());
+    throw std::runtime_error(errMsg);
+  }
```

Comment:
This looks odd to me.  Why is translateSectionKindStrToKind returning an error code that has to be checked as opposed to itself throwing on error and return kind on success?

Reply:
Agree, the existing code was odd. 

I have refactored the code base so that `translateSectionKindStrToKind()` now throws an exception and no longer returns a boolean value.   The code is indeed cleaner.